File: backend/app/routes/waitlist.py
from fastapi import APIRouter

from ..schemas.waitlist import WaitlistSignupRequest, WaitlistSignupResponse
from ..services.waitlist_service import WaitlistService

router = APIRouter(prefix='/api/waitlist', tags=['waitlist'])


@router.post('/signup', response_model=WaitlistSignupResponse)
async def waitlist_signup(payload: WaitlistSignupRequest) -> WaitlistSignupResponse:
    result = await WaitlistService().signup(
        str(payload.email),
        parent_name=payload.parent_name,
        child_grade=payload.child_grade,
        interest_note=payload.interest_note,
    )
    return WaitlistSignupResponse(**result)

File: frontend/src/lib/api/waitlist.ts
import { apiPost } from '../api';

export type WaitlistSignupResponse = {
  success: boolean;
  message: string;
};

export type WaitlistSignupPayload = {
  parent_name?: string;
  email: string;
  child_grade?: string;
  interest_note?: string;
};

export function joinWaitlist(payload: WaitlistSignupPayload): Promise<WaitlistSignupResponse> {
  return apiPost<WaitlistSignupResponse>('/api/waitlist/signup', payload);
}
